aibolit/patterns/classic_setter/classic_setter.py

```diff
@@ -13,22 +11,18 @@ def __init__(self):
 
     def value(self, filename: str) -> List[LineNumber]:
         lst: List[LineNumber] = []
-        tree = build_ast(filename).filter(javalang.tree.MethodDeclaration)
-        for _, node in tree:
-            if (node.return_type is None) and ('set' in node.name[:3]):
-                if (isinstance(node.body, list)) and len(node.body) < 2:
-                    for statement in node.body:
-                        if isinstance(statement, javalang.tree.StatementExpression):
-                            if isinstance(statement.expression, javalang.tree.Assignment):
-                                expression = statement.expression.expressionl
-                                if isinstance(expression, javalang.tree.This):
-                                    if statement.expression.type == '=':
-                                        if expression.selectors[0].member.lower() == node.name.lower()[3:]:
-                                            lst.append(node._position.line)
-                                    else:
-                                        break
-                                else:
-                                    break
-                            else:
-                                break
+        ast = JavaPackage(filename)
+        method_decls = list(ast.get_nodes(ASTNodeType.METHOD_DECLARATION))
+        for node in method_decls:
+            method_name = ast.get_attr(node, 'name')
+            body_nodes = ast.get_attr(node, 'body')
+            if (ast.get_attr(node, 'return_type') is None) and method_name.startswith('set') and len(body_nodes) < 2:
```

Comment:
@Vitaly-Protasov if can't avoid MagicNumber `2`, could we left a note why we use it?

Reply:
okay!